### DISCUSSION TITLE
It is not possible to create a custom large control.

### DISCUSSION THREAD

**Discussion by korenevskiy** [joomla/joomla-cms#32604] [General]
![img-2021-03-07-13-01-04](https://user-images.githubusercontent.com/6898474/110236149-48748980-7f45-11eb-9199-66ddfd85a6fc.png)
It is not possible to create a custom large control.
The layout of the field module settings is distributed across the width of the screen by columns.
That's a good idea.
On the 1920*1080 screen, the fields are arranged in 3 columns.
But why is this done using the CSS property column-count???
This is a bad idea.!!!
And what if the field is not just a text block, but a metro diagram, or an editor, or a table?
The width of such controls does not fit into the width of 500px.
By specifying the column-count property, developers restrict freelancers and companies to use Joomla for their needs.
Joomla should not limit the size of the control to 500px, which means that the 3-column layout should be created in a different way.
For example, GRID, FLEX, ~~FLOAT:left;~~
~~CSS property column-count is even worse than FLOAT:left;~~ 
Let's think about how to make a 3-column layout, but so that it does not limit the conrolls with a large width.

**Comment by korenevskiy** [comment]
I suggest making the XML **`<fieldset>`** tag configurable.
By default, it will work as is.
But so that you can make the Fieldset initially **single-column**.
Specifying **column-count**.
**Example:**
**mod_usermodule.xml**
```
<fieldset  name="user_tab" label="Table Fields" column-count="1">
    <field name="name1" type="text"/>
    <field name="name2" type="table"/>
</fieldset>
```

**Comment by Fedik** [comment]
use `groupByFieldset="true"` https://docs.joomla.org/Subform_form_field_type look for "An example XML of exampleform.xml with fieldsets"

**Reply by korenevskiy** [reply]
there is little information there. unclear.
Please provide an XML example.

**Reply by Fedik** [reply]
field:
```xml
<field name="field-name" type="subform"
  formsource="path/to/exampleform.xml" 
  multiple="true"
  groupByFieldset="true"
  layout="joomla.form.field.subform.repeatable-table"
  label="Subform Field" />
```
`exampleform.xml` : 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<form>
    <fieldset name="section1" label="Section1">
        <field name="example_text" type="text" label="Example Text" />
        <field name="example_textarea" type="textarea" label="Example Textarea" cols="40" rows="8" />
    </fieldset>
    <fieldset name="section2" label="Section2">
        <field name="example_list" type="list" default="1" class="advancedSelect" label="Example List">
            <option value="1">JYES</option>
            <option value="0">JNO</option>
        </field>
    </fieldset>
</form>
```

**Reply by korenevskiy** [reply]
![img-2021-03-07-18-32-38](https://user-images.githubusercontent.com/6898474/110245277-8ab4bf80-7f73-11eb-8d4f-bb79958d4dbb.png)


Thank you for the example.
But what you wrote has nothing to do with my question at all. Either I didn't explain it well, or you didn't understand it well.
.
Each field in the admin panel is located in the `<div class="control-group">`tag.
The tag container has the column-count=3 style.
If there are many fields, the **<div class="control-group">** tags are arranged in 3 columns each with a width of 550px.
Even if the width of the control is larger, it will still be in the column with 550px. At the same time, climbing on another column in which there are other controls.
.
In the screenshot, do you see the conrolls in 3 columns?
Why do you think they are in 3 columns and not in 4?
They're all on top of each other!

**Comment by Quy** [comment]
Please test using the nightly builds: https://developer.joomla.org/nightly-builds.html

See https://github.com/joomla/joomla-cms/pull/32422

**Comment by drmenzelit** [comment]
Please test #32488
